app/data/schema/schema-v1.json

```diff
@@ -2,6 +2,10 @@
   "$schema": "http://json-schema.org/draft-04/schema#",
   "type": "object",
   "definitions": {
+    "id": {
+        "type": "string",
+        "pattern": "^[a-z0-9][a-z0-9\\-]*[a-z0-9]$"
```

Comment:
The GUIDs satisfy this regex so this won't break on account of the test schemas. I think the main purpose of adding this is to ensure that nobody adds any stupid identifiers in the future.
One change I think should be made in the future (once all the test schemas have been converted) is to drop the 0-9 from the first match. This will help to enforce that Ids don't start with a number. But GUIDS will break that at the moment.

I tested the RegExp using an online tool, but if it's helpful for documentation I can also write a few tests.